### DISCUSSION TITLE
disallowing unknown attributes

### DISCUSSION THREAD

**Discussion by adityasarawatexponential** [json-schema-org/community#174] [Q&A] [answered]
I want to prevent the attributes that are not present in schema.

**Comment by jviotti** [comment, marked as answer]
@Relequestual Would it make sense to offer a "strict" version of the official meta-schemas? I found myself looking for this now and then, as its very easy to make a typo, etc without realizing.

**Reply by handrews** [reply]
@adityasarawatexponential the `unevaluatedProperties` keyword was added to solve this problem.  The idea of requiring implementations to support a "strict mode" was one of the controversies that led to the near-collapse of the project after draft-04 (none of the people directly involved in that debate are still with the project).  At least one major implementation has such a mode and from what I can tell it also causes confusion.

@jviotti one of my motivations for adding both `unevaluatedProperties` and what eventually became `$dynamicRef` is that with those two keywords it's trivial to make a strict meta-schema.

@Relequestual would "how to make a strict meta-schema" be a good blog post?  I could write that one pretty quickly I think, just as a straightforward tutorial.  Not an in-depth exploration of theory 😜
